Mage/src/main/java/mage/filter/predicate/permanent/EnchantedByPlayerPredicate.java

```diff
@@ -0,0 +1,35 @@
+package mage.filter.predicate.permanent;
+
+import mage.constants.SubType;
+import mage.filter.predicate.Predicate;
+import mage.game.Game;
+import mage.game.permanent.Permanent;
+
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * @author Xanderhall
+ */
+public class EnchantedByPlayerPredicate implements Predicate<Permanent> {
```

Comment:
This could be an enum with a single instance (and probably need another name then).
I would go with `public enum ... implements ObjectSourcePlayerPredicate<Permanent>`.

Then in the apply you have access to `ObjectSourcePlayer<Permanent> input`, which allows to have the source's controller to compare with the Aura's controller.

Reply:
I see!